### DISCUSSION TITLE
[Help Wanted] How to share subscription with Private Username

### DISCUSSION THREAD

**Discussion by jmason888** [standardnotes/forum#2674] [Suggestions]
**Describe the issue**
I don't see a way to share my subscription with a private username

**To Reproduce**
1. Go to 'Account Settings'
2. Scroll to Subscription Sharing
3. Invite
4. System asks for "Invitee Email"
5. Private Usernames _don't have_ an email address

**Expected behavior**
1. Allow the user to enter a Private Username (or the related hash) in this box
2. Give the user the invite link *directly* rather than emailing it (since the invitee has no email address)

**Additional context**
If you can think of any other way to do subscription sharing between a regular account and a private username, that would work too.

**Comment by effieeee** [comment]
Indeed we still need to work on having subscription sharing work for private workspaces 😞We're keeping this in mind!

**Comment by pwseo** [comment]
@effieeee is this on your plans for the future? It is indeed a very useful feature, because one often uses private usernames as a way to divide notes into separate namespaces / workspaces.

**Comment by effieeee** [comment]
Hi @pwseo, while it's not on the current priorities yet, this is still in consideration, since the subscription sharing feature should also understandably be applicable for private workspaces too 👍

**Comment by jmason888** [comment]
Is this still under consideration?

This seems like it shouldn't be too hard to solve.  Below I'm thinking of the general case where the recipient and the sender are not the same entity; the use case that we are talking about ,where an SN subscriber wants to share with _their own_ private workspace, is really a special case of this.

- In the "share subscription" UI, allow the subscriber to input of the recipient's hashed username instead of their email. 
  - This would be the easiest thing to do, since SN uses that hash in place of an email internally.
  - A "private username"  option like in the sign-in flow wouldn't hurt, which does the hashing internally. But perhaps still the input  of the raw username hash, for cases where the recipient doesn't want to reveal their private username.
- Instead of emailing the invitation code, have an option to simply display that code to the subscriber who can transmit the invitation to the indented recipient by other means. 
   - This also would be helpful for cases where the recipient would rather communicate using a secure method like Signal


Together those two UI changes would allow a subscriber to share their subscription with their own private workspace.  And it would definitely enforce the "must have an existing SN account" requirement, since the recipient would normally have to create a private workspace in order to get the hash they need.
